sr_health_check/src/sr_health_check/sr_health_check_gui.py

```diff
@@ -169,31 +171,34 @@ def check_execution(self):
             check['thread'].join()
 
             self._check_queue.task_done()
-            self._results[self._filename].append(check['check'].get_result())
+            self._results[self._entry_name].update(check['check'].get_result())
             self.update_passed_label(check['check'])
 
             if self._check_queue.empty():
-                file = f"{rospkg.RosPack().get_path('sr_health_check')}/src/sr_health_check/results.yaml"
-                if exists(file):
-                    with open(file, 'r', encoding="ASCII") as yaml_file:
-                        self._current_data = yaml.safe_load(yaml_file)
-                    with open(file, 'w', encoding="ASCII") as yaml_file:
-                        if self._current_data:
-                            self._current_data.update(self._results)
-                        else:
-                            self._current_data = dict(self._results)
-                        yaml.safe_dump(self._current_data, stream=yaml_file, default_flow_style=False)
-                else:
-                    with open(file, 'w', encoding="ASCII") as yaml_file:
-                        self._current_data = dict(self._results)
-                        yaml.safe_dump(self._current_data, stream=yaml_file, default_flow_style=False)
+                self.save_results_to_result_file(self._results)
+
+    def get_data_from_results_file(self):
+        output = {}
+        try:
+            with open(self._file, 'r', encoding="ASCII") as yaml_file:
+                output = yaml.safe_load(yaml_file) or {}
+        except FileNotFoundError:
+            rospy.logwarn("Result file does not exists. Returning empty dictionary")
+        return output
+
+    def save_results_to_result_file(self, results):
+        if results:
+            self._current_data.update(self._results)
+        if exists(self._file):
```

Comment:
If the results.yaml file doesn't exist, Shouldn't we create if from self._current_data?